src/riak_kv_group_keys.erl

```diff
@@ -0,0 +1,547 @@
+        undefined ->
+            Default;
+        Value ->
+            Value
+    end.
+
+to_binary(undefined) ->
+    undefined;
+to_binary(Param) when is_list(Param) ->
+    list_to_binary(Param);
+to_binary(Param) when is_binary(Param) ->
+    Param.
+
+to_integer(undefined) ->
+    undefined;
+to_integer(Param) when is_list(Param) ->
+    list_to_integer(Param);
+to_integer(Param) when is_binary(Param) ->
+    binary_to_integer(Param);
+to_integer(Param) when is_integer(Param) ->
+    Param.
+
+get_backend(#group_params{backend_mod = BackendMod}) ->
+    BackendMod.
+
+get_prefix(#group_params{prefix = Prefix}) ->
+    Prefix.
+
+get_delimiter(#group_params{delimiter = Delimiter}) ->
+    Delimiter.
+
+get_start_after(#group_params{start_after = StartAfter}) ->
+    StartAfter.
+
+get_max_keys(#group_params{max_keys = MaxKeys}) ->
+    MaxKeys.
+
+get_continuation_token(#group_params{continuation_token = ContinuationToken}) ->
+    ContinuationToken.
+
+-spec set_prefix(group_params(), Prefix::binary()) -> group_params().
+set_prefix(GroupParams, Prefix) ->
+    GroupParams#group_params{prefix = Prefix}.
+
+-spec set_delimiter(group_params(), Delimiter::binary()) -> group_params().
+set_delimiter(GroupParams, Delimiter) ->
+    GroupParams#group_params{delimiter = Delimiter}.
+
+-spec set_start_after(group_params(), StartAfter::binary()) -> group_params().
+set_start_after(GroupParams, StartAfter) ->
+    GroupParams#group_params{start_after = StartAfter}.
+
+-spec set_max_keys(group_params(), MaxKeys::pos_integer()) -> group_params().
+set_max_keys(GroupParams, MaxKeys) ->
+    GroupParams#group_params{max_keys = MaxKeys}.
+
+-spec set_continuation_token(group_params(), ContinuationToken::binary()) ->
+    group_params().
+set_continuation_token(GroupParams, ContinuationToken) ->
+    GroupParams#group_params{continuation_token = ContinuationToken}.
+
+%% NOTE: This  type spec with `fun(() -> no_return())' as the return type is
+%% necessary to avoid a Dialyzer warning.
+-spec content_folder_fun(any(), group_params(), any(), any(), any(), any()) ->
+    fun(() -> no_return()).
+content_folder_fun(Bucket, GroupParams, DbRef, FoldOpts, FoldFun, Acc) ->
+    fun() ->
+            content_folder(Bucket, GroupParams, DbRef, FoldOpts, FoldFun, Acc)
+    end.
+
+%% NOTE: This  type spec with `no_return()' as the return type is necessary to
+%% avoid a Dialyzer warning.
+-spec content_folder(any(), group_params(), any(), any(), any(), any()) ->
+    no_return().
+content_folder(Bucket, GroupParams, DbRef, FoldOpts, FoldFun, Acc) ->
+    BackendMod = get_backend(GroupParams),
+    FoldOpts1 = [{first_key, BackendMod:to_first_key({bucket, Bucket})} | FoldOpts],
+    try iterator_open(DbRef, FoldOpts1) of
+        {ok, Itr} ->
+            iterate(Bucket, GroupParams, Itr, FoldFun, Acc)
+    catch Error ->
+              lager:debug("Could not open iterator: ~p", [Error]),
+              throw(Error)
+    end.
+
+iterator_open(DbRef, FoldOpts) ->
```

Comment:
Could we maybe get rid of this extra layer of indirection for `iterator_open`, `iterator_close`, and `iterator_move`? Feels kind of like overkill just to avoid having to type "eleveldb:" in a few places.